scalr/utils/data_utils.py

```diff
@@ -82,3 +83,62 @@ def generate_dummy_anndata(n_samples, n_features, target_name='celltype'):
     adata.obs.index = adata.obs.index.astype(str)
 
     return adata
+
+
+def generate_dummy_dge_anndata(
+        n_donors: int = 5,
+        cell_type_list: list[str] = ['B_cell', 'T_cell', 'DC'],
+        cell_replicate: int = 2,
+        n_vars: int = 10):
+    """This function returns anndata object for DGE analysis
+    with shape (n_donors*len(cell_type_list)*cell_replicate, n_vars).
+
+    It generates obs with random donors with a fixed clinical condition (disease_x or normal).
+    Includes all the celltypes in cell_type_list with no. of cell_replicate for each donor.
+    It generates a csr(Compressed Sparse Row) matrix with random gene expression values.
+    It generates var with random gene_name as var.index of length `n_vars` .
+
+    Args:
+        n_donors: Number of donors or subjects in anndata.obs.
+        cell_type_list: List of different cell types to include.
+        cell_replicate: Number of cell replicates per cell type.
+        n_vars: Number of genes to include in anndata.var.
+
+    Returns:
+        Anndata object.
+    """
+
+    # Setting seed for reproducibility.
+    np.random.seed(0)
+
+    donor_list = [f'D_{i}' for i in range(1, n_donors + 1)]
+    condition_array = np.random.choice(['disease_x', 'normal'],
+                                       size=n_donors,
+                                       replace=True)
+
+    # Creating obs
+    obs_data = []
+    for donor, condition in zip(donor_list, condition_array):
+        obs_data.extend([{
+            'donor_id': donor,
+            'cell_type': cell_type,
+            'disease': condition
+        } for i in range(cell_replicate) for cell_type in cell_type_list])
+    obs = pd.DataFrame(obs_data)
+    n_obs = obs.shape[0]
+    obs.index = obs.index.astype(str)
+
+    # Random geneexpression matrix
+    X = csr_matrix(np.random.rand(n_obs, n_vars))
+
+    # Creating var
+    var = pd.DataFrame({
+        'gene_id': [f'gid_{i}' for i in range(1, n_vars + 1)],
+        'gene_name': [f'G{i}' for i in range(1, n_vars + 1)]
```

Comment:
```suggestion
        'gene_name': [f'Gene_{i}' for i in range(1, n_vars + 1)]
```

Reply:
Changed from 'G_i' to 'gene_i' for gene names.